main/python-cffi/patches/clang-errors.patch

```diff
@@ -0,0 +1,24 @@
+Based on https://github.com/python-cffi/cffi/pull/78, but also performing a cast
+to size_t to avoid errors like:
+
+_CFFI_test_verify_anonymous_struct_with_star_typedef.cpp:583:10: error: non-constant-expression cannot be narrowed from type 'long' to 'size_t' (aka 'unsigned long') in initializer list [-Wc++11-narrowing]
+  583 |   { "a", ((char *)&((foo_t)4096)->a) - (char *)4096,
+      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+_CFFI_test_verify_anonymous_struct_with_star_typedef.cpp:583:10: note: insert an explicit cast to silence this issue
+  583 |   { "a", ((char *)&((foo_t)4096)->a) - (char *)4096,
+      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+      |          static_cast<size_t>(                      )
+
+diff --git a/src/cffi/recompiler.py b/src/cffi/recompiler.py
+index 4167bc05..ac6c163e 100644
+--- a/src/cffi/recompiler.py
++++ b/src/cffi/recompiler.py
+@@ -953,7 +953,7 @@ def _struct_ctx(self, tp, cname, approxname, named_ptr=None):
+                 if cname is None or fbitsize >= 0:
+                     offset = '(size_t)-1'
+                 elif named_ptr is not None:
+-                    offset = '((char *)&((%s)0)->%s) - (char *)0' % (
++                    offset = '(size_t)(((char *)&((%s)4096)->%s) - (char *)4096)' % (
```

Comment:
i _think_ this is correct c++ code that behaves the same as the original, but do let me know if i'm wrong

Reply:
it should probably be just `offsetof(%s, %s)`

Reply:
for posterity, `offsetof` wouldn't work here: https://github.com/python-cffi/cffi/pull/78#issuecomment-2198107460